src/redux/cartReducer/cartReducer.js

```diff
@@ -0,0 +1,49 @@
+import produce from "immer";
+import { ADD_ITEM_IN_CART, REMOVE_ITEM_IN_CART } from "./cartActionTypes";
+
+const cartState = {
+  cartData: new Map(),
+  productData: new Map(),
+};
+
+const cartReducer = (state = cartState, action) => {
+  switch (action.type) {
+    case ADD_ITEM_IN_CART: {
+      let product = action.payload.product;
+
+      const id = product["product__id"];
+
+      return produce(state, (draft) => {
+        if (draft.productData.get(id)) {
+          let count = parseInt(draft.productData.get(id));
+          draft.productData.set(id, count + 1);
+        } else {
+          draft.cartData.set(id, product);
+          draft.productData.set(id, 1);
+        }
+      });
+    }
+
+    case REMOVE_ITEM_IN_CART: {
+      let product = action.payload.product;
```

Comment:
Safety check is missing. Please read about optional chaining and get function of lodash